File: 2.js
var prev = 1;
var cur = 2;
var sum = 0;

while(cur < 4000000){
    if(cur % 2 === 0){
        sum += cur;
    }

    var temp = prev;
    prev = cur;
    cur += temp;
}

console.log(sum);

File: 2.cpp
#include <iostream>
using namespace std;

int main() {
  int prev = 1;
  int cur = 2;
  int sum = 0;

  while(cur < 4000000) {
    if(cur % 2 == 0) {
      sum += cur;
    }

    int temp = prev;
    prev = cur;
    cur += temp;
  }

  cout << sum << endl;
}
